bin/ruby-build

```diff
@@ -793,11 +793,21 @@ remove_windows_files() {
   rm -f bin/*.exe bin/*.dll bin/*.bat bin/jruby.sh
 }
 
+check_path_is_empty() {
+  shopt -s nullglob dotglob
+  local files_array=(${1}/*)
+  local is_empty=${#files_array[*]}
+  shopt -u nullglob dotglob
+  return $is_empty
```

Comment:
I think this solution feels rather cryptic. Googling a bit for "bash check directory empty" I get https://superuser.com/questions/352289/bash-scripting-test-for-empty-directory.
I think the `if [ -z "$(ls -A /path/to/dir)" ]; then` solution is much simpler, easier to read, and likely also more portable.

Could you switch to that?

Reply:
I agree it's not the most straightforward one; I guess I misinterpreted your request for "in bash" to mean using bash built-ins rather than just a standard POSIX shell environment. 

The array functionality involved has been in bash for a long time and is apparently portable to most ksh-like shells; it just expands a glob into an array and counts the members using the standard indexed-array functionality. But arrays are definitely not a well-known or POSIX-specified part of bash and I had to look them up to understand how it worked.

I'll happily switch to the ls version though; will update the PR in a bit.